gmd-sdk/nifi-data-processors/src/main/scala/com/deciphernow/greymatter/data/nifi/processors/GetPolicies.java

```diff
@@ -257,84 +249,17 @@ public void setUpClient(final ProcessContext context) throws IOException, Unreco
         // Set whether to follow redirects
         okHttpClientBuilder.followRedirects(context.getProperty(PROP_FOLLOW_REDIRECTS).asBoolean());
 
+        // Apply the TLS configuration if present
         final SSLContextService sslService = context.getProperty(PROP_SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
-        final SSLContext sslContext = sslService == null ? null : sslService.createSSLContext(ClientAuth.NONE);
-
-        // check if the ssl context is set and add the factory if so
-        if (sslContext != null) {
-            setSslSocketFactory(okHttpClientBuilder, sslService, sslContext);
+        if (sslService != null) {
+            final TlsConfiguration tlsConfiguration = sslService.createTlsConfiguration();
+            OkHttpClientUtils.applyTlsToOkHttpClientBuilder(tlsConfiguration, okHttpClientBuilder);
```

Comment:
In same PR the OkHttpClientUtils https://github.com/apache/nifi/commit/441781cec50f77d9f1e65093f55bbd614b8c5ec6#diff-7251dc1adc04dc044eec2b217c8e3b70d84b6331b153f98f8520aa39829ea93c

Reply:
It wraps this 
```
final SSLSocketFactory sslSocketFactory = SslContextFactory.createSSLSocketFactory(tlsConfiguration);
```

Reply:
It should be initializing the keystore with keystorepassword + keypassword or just keystorepassword here https://github.com/apache/nifi/commit/441781cec50f77d9f1e65093f55bbd614b8c5ec6#diff-2876b4f0115a43570c21c95e516760187b37ea5db3990ea0a12f9056ccf01a8eR133